// ceymikey-expansion/src/index.ts
// src/index.ts
export const printOut = (): void => {
  console.log('Test printOut succes');
};

export const executeConsoleString = (consoleString: string): void => {
  console.log(`Constole string executed with succes`);
};

//* NEW INTERFACE FOR `ceymikey.compile`
interface Logger {
  log(message: string): void;
}

export const thislogger: Logger = {
  log: (message: string): void => {
    console.log(message);
  }
};

export const compile = (input: { data: { fetch: (params: { cout: { controle: () => string } }) => void } }): void => {
  try {
    const { data } = input;

    if (!data || !data.fetch || typeof data.fetch !== 'function') {
      throw new Error('Invalid data structure: "fetch" function is missing or not a function.');
    }

    const cout = data.fetch({ cout: { controle: () => 'custom string here' } });
    if (typeof cout !== 'string') {
      throw new Error('Invalid data structure: "cout" is not a string.');
    }

    thislogger.log(cout);
  } catch (error) {
    console.error(`Error in ceymikey.compile: ${error}`);
  }
};
