// ceymikey-expansion/dist/index.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.compile = exports.thislogger = exports.executeConsoleString = exports.printOut = void 0;
// src/index.ts
const printOut = () => {
    console.log('Test printOut succes');
};
exports.printOut = printOut;
const executeConsoleString = (consoleString) => {
    console.log(`Constole string executed with succes`);
};
exports.executeConsoleString = executeConsoleString;
exports.thislogger = {
    log: (message) => {
        console.log(message);
    }
};
const compile = (input) => {
    try {
        const { data } = input;
        if (!data || !data.fetch || typeof data.fetch !== 'function') {
            throw new Error('Invalid data structure: "fetch" function is missing or not a function.');
        }
        const cout = data.fetch({ cout: { controle: () => 'custom string here' } });
        if (typeof cout !== 'string') {
            throw new Error('Invalid data structure: "cout" is not a string.');
        }
        exports.thislogger.log(cout);
    }
    catch (error) {
        console.error(`Error in ceymikey.compile: ${error}`);
    }
};
exports.compile = compile;
